driver-core/src/test/java/com/datastax/driver/core/PercentileTrackerTest.java

```diff
@@ -0,0 +1,166 @@
+
+        // when - recording less measurements then required.
+        long startTime = System.currentTimeMillis();
+        for (int i = 0; i < 99; i++) {
+            tracker.update(defaultHost, defaultStatement, defaultException, TimeUnit.NANOSECONDS.convert(1, TimeUnit.MILLISECONDS));
+        }
+        long waitTime = 50 - (System.currentTimeMillis() - startTime);
+        Uninterruptibles.sleepUninterruptibly(waitTime + 100, TimeUnit.MILLISECONDS);
+
+        // then - the resulting tracker's percentiles should all be -1, indicating there were not enough values to consider.
+        for (int i = 1; i <= 99; i++) {
+            long latencyAtPct = tracker.getLatencyAtPercentile(defaultHost, defaultStatement, defaultException, i);
+            assertThat(latencyAtPct).isEqualTo(-1);
+        }
+    }
+
+    @Test(groups = "short")
+    public void should_return_negative_value_when_interval_hasnt_elapsed() throws Exception {
+        // given - a percentile tracker with a long interval.
+        Cluster cluster0 = mock(Cluster.class);
+        T tracker = builder()
+                .withInterval(50, TimeUnit.MINUTES)
+                .withMinRecordedValues(100).build();
+        tracker.onRegister(cluster0);
+
+        // when - recording enough measurements.
+        for (int i = 0; i < 99; i++) {
+            tracker.update(defaultHost, defaultStatement, defaultException, TimeUnit.NANOSECONDS.convert(1, TimeUnit.MILLISECONDS));
+        }
+
+        // then - the resulting tracker's percentiles should all be -1, since not enough time was given to elapse interval.
+        for (int i = 1; i <= 99; i++) {
+            long latencyAtPct = tracker.getLatencyAtPercentile(defaultHost, defaultStatement, defaultException, i);
+            assertThat(latencyAtPct).isEqualTo(-1);
+        }
+    }
+
+    @Test(groups = "unit")
+    public void should_not_record_value_and_log_when_measurement_higher_than_max_trackable_value() throws Exception {
+        // given - a percentile tracker with a long interval.
+        Cluster cluster0 = mock(Cluster.class);
+        T tracker = builder()
+                .withInterval(50, TimeUnit.MILLISECONDS)
+                .withMinRecordedValues(100).build();
+        tracker.onRegister(cluster0);
+
+        Logger percentileLogger = Logger.getLogger(PercentileTracker.class);
+        Level originalLevel = percentileLogger.getLevel();
+        percentileLogger.setLevel(Level.WARN);
+        MemoryAppender appender = new MemoryAppender();
+        percentileLogger.addAppender(appender);
+
+        try {
+            long startTime = System.currentTimeMillis();
+            for (int i = 0; i < 100; i++) {
+                tracker.update(defaultHost, defaultStatement, defaultException, TimeUnit.NANOSECONDS.convert(1, TimeUnit.MILLISECONDS));
+            }
+
+            // HdrHistogram adjusts its max based on byte count, with a max of 1000, that allows a max vaue of 2047 (2^8-1).
```

Comment:
my math is a bit off there :)  (should be 2^11-1).  Curious as to why setting a max to 1000 ends up allowing to 2047 though, will look into it, probably some implementation detail in hdrhisto